aiobotocore/retries/bucket.py

```diff
@@ -0,0 +1,109 @@
+"""An async reimplementation of the blocking elements from botocore.retries.bucket."""
+import asyncio
+
+from botocore.exceptions import CapacityNotAvailableError
+from botocore.retries.bucket import Clock as Clock  # reexport # noqa
+
+
+class AsyncTokenBucket:
```

Comment:
why no inherit? otherwise we won't pick up any methods they later add

Reply:
The rationale was (I'll add a comment to this effect if you're OK with that):

The interface is different from TokenBucket. In TokenBucket, `max_rate` also has a setter. In `AsyncTokenBuket`, I switched to `async set_max_rate`. If using inheritance, we would have to override the setter to a method that raises an exception, since we no longer want anyone to use this setter.

I'm not sure we want to pick up automatically new methods here, as they may be blocking, and it's simpler to understand what's going on if the code fails with an AttributeError rather than if the event loop blocks in production.

Reply:
that's fine if you leave a comment to this effect, however this also means in patches you should set the patch on the class itself and not the methods.  That way if anything changes in the class it will trigger and someone needs to look.   Please also put cmt in patch for them to look in the comment you'll add in AsyncTokenBucket.